src/script/standard.cpp

```diff
@@ -35,22 +35,55 @@ const char* GetTxnOutputType(txnouttype t)
     return nullptr;
 }
 
-bool Solver(const CScript& scriptPubKey, txnouttype& typeRet, std::vector<std::vector<unsigned char> >& vSolutionsRet)
+static bool MatchPayToPubkey(const CScript& script, valtype& pubkey)
 {
-    // Templates
-    static std::multimap<txnouttype, CScript> mTemplates;
-    if (mTemplates.empty())
-    {
-        // Standard tx, sender provides pubkey, receiver adds signature
-        mTemplates.insert(std::make_pair(TX_PUBKEY, CScript() << OP_PUBKEY << OP_CHECKSIG));
+    if (script.size() == 35 && script[0] == 33 && script[34] == OP_CHECKSIG) {
+        pubkey = valtype(script.begin() + 1, script.begin() + 34);
+        return CPubKey::ValidSize(pubkey);
+    }
+    if (script.size() == 67 && script[0] == 65 && script[66] == OP_CHECKSIG) {
+        pubkey = valtype(script.begin() + 1, script.begin() + 66);
+        return CPubKey::ValidSize(pubkey);
+    }
+    return false;
+}
+
+static bool MatchPayToPubkeyHash(const CScript& script, valtype& pubkeyhash)
```

Comment:
See also `IsToKeyID` in compressor.cpp, which looks equivalent.

Could split this up as `CScript::IsPayToPubkeyHash`, and do the unpacking in `Solver`, ala `IsPayToScriptHash`. Would make the case handling more consistent in Solver.